gulpfile.js

```diff
@@ -56,54 +56,57 @@ gulp.task('example-serve', ['browserify', 'styles', 'watch'], function() {
     });
 });
 
-gulp.task('tests-browserify', function() {
-    return browserify('./test/tests')
-        .bundle()
-        .on('error', function (err) {
-            console.log(err.toString());
-            this.emit('end');
-        })
-        .pipe(source('tests.js'))
-        .pipe(gulp.dest('test/build/'));
-});
-
-gulp.task('test', ['tests-browserify'], function () {
-    var browserTests = gulp
-    .src('./test/test-runner.html')
-    .pipe(mochaPhantomJS({
+function gulp_test() {
+    return gulp.src([
+        'test/**/*.spec.js'
+    ])
+    .pipe(mocha({
         log: true,
         timeout: 10000,
         slow: 3000,
         reporter: 'spec',
-        ui: 'bdd'
+        ui: 'bdd',
+        require: ['./test/init.js']
     }));
+}
 
-    var nodeTests = gulp
-    .src([
-        'test/lib/**/*.spec.js',
-
-        // exclude lib browser tests
-        '!test/lib/charts/*.spec.js',
-        '!test/lib/generators/*.spec.js',
-        '!test/lib/components/*.spec.js'
+gulp.task('instrument', function () {
+    return gulp.src([
+        'src/**/*.js',
+        '!src/lib/components/log-explorer/**/*.js'
```

Comment:
I'm guessing this is here cause isparta or someone else isn't happy with jsx? This prompted me to wonder if its more correct to run tests against the original code or transpiled code in `lib` (a la [this stackoverflow question](http://stackoverflow.com/questions/33228754/in-a-babel-project-workflow-does-it-make-more-sense-to-run-unit-tests-firstly-a)). From a quick look at two libraries I've been using recently, [react-select](https://github.com/JedWatson/react-select) and [redux](https://github.com/rackt/redux), both test against the original code. Don't have a strong preference either way.